src/Illuminate/Foundation/Providers/FoundationServiceProvider.php

```diff
@@ -70,7 +71,7 @@ public function boot()
             ], 'laravel-errors');
         }
 
-        if ($this->app->hasDebugModeEnabled()) {
+        if ($this->app->hasDebugModeEnabled() && ! app()->has(ExceptionRenderer::class)) {
```

Comment:
Why use the `app()` helper here instead of the `$this->app` property, as it is used before the `&&` operator?